core/soc/estimator.go

```diff
@@ -193,7 +193,7 @@ func (s *Estimator) SoC(chargedEnergy float64) (float64, error) {
 			// sample charged energy at soc change, reset energy delta
 			s.prevChargedEnergy = math.Max(chargedEnergy, 0)
 			s.prevSoc = s.vehicleSoc
-		} else {
+		} else if fetchedSoC != nil {
```

Comment:
Kann der hier `nil` sein?
Das wird doch weiter oben schon abgefangen.

Reply:
Nein kann im aktuellen Source nicht "nil" sein, wird oben "implizit" abgefangen. Die Frage ist wie ihr es haben wollt. Ich kenne es so das man Pointer explizit prüft. Mir ist das beim Refactoring sofort gecrashed, weil ich oben was geändert hatte. Frage: Implizit vs explizit. Aktuell kein Bug und eine Statische Sourcecodeanalyse würde auch nicht meckern.

Reply:
Wenns nicht passieren kann brauchts keinen Fix. Im Zweifel kaschier die Änderung dann einen anderen Bug in der Logik...

Reply:
Ich hätte es nicht besser ausdrücken können. 😁

Der Estimator-Code ist inzwischen einerseits relativ ausgereift, andererseits aber auch empfindlich, da viele Randszenarien beachtet werden müssen.

Änderungen daran würde ich daher immer nur wohlüberlegt und anschließend wirklich sehr gut getestet wissen wollen.

Reply:
Das schließt natürlich nicht aus dass es vielleicht auch doch noch irgendwo etwas zu verbessern, verschönern oder zu optimieren gäbe.

Reply:
Danke für die Blumen. Dann bin ich bei euch und wir lassen es so. Danke fürs Prüfen.